### DISCUSSION TITLE
How to handle Unique Constrains

### DISCUSSION THREAD

**Discussion by GewoonJaap** [pocketbase/pocketbase#1438] [Q&A] [answered]
Hi,
I have a likedPost collection with a user reference and a post reference:
![afbeelding](https://user-images.githubusercontent.com/33700526/210063894-83d050fe-257d-41f9-99ab-0135736ac346.png)
![afbeelding](https://user-images.githubusercontent.com/33700526/210064035-3c0cdce5-4d2d-496a-904c-3d673ae20478.png)

I want that a user can only like the post once, I tried adding Unique to both fields, but then a user can only like once, and a post can only be liked once. Because the user ID and post ID can only be added to this collection once. But I wanted a unique combination of User and Post.

How to achieve this? If this needs to be done with a rule, thats fine

**Comment by ganigeorgiev** [comment, marked as answer]
Unique constraint across multiple fields currently is not supported from the UI. It will be handled in https://github.com/pocketbase/pocketbase/issues/544.

The easiest workaround for now would be add a unique index to your database manually or via js/go migration, eg.:

- manually via the sqlite3 cli tool:
    ```sh
    sqlite3 ./pb_data/data.db 'CREATE UNIQUE INDEX likedPost_liker_post_idx ON likedPost (liker, post)'
    ```

- via JS (or Go) migration (recommended):
    1. Run `./pocketbase migrate create 'add_likedPost_index'`
    2. Navigate to the created file in your `pb_migrations` and add the following:
       ```js
       migrate((db) => {
         // up
         db.createUniqueIndex("likedPost", "likedPost_liker_post_idx", "liker", "post").execute()
       }, (db) => {
         // down
         db.dropIndex("likedPost", "likedPost_liker_post_idx").execute()
       })
       ```
       
       _(the migration will be applied automatically on the next `serve`; this option has the benefit that you can commit the migration file in your repo)_

**Reply by GewoonJaap** [reply]
Ah very cool! Thank you so much :) Your tool works very good

**Reply by ganigeorgiev** [reply]
@GewoonJaap _One small update to the above migration snippet - `createIndex` should be `createUniqueIndex` for a unique index constraint. If you have already executed the above migration, you can run `./pocketbase migrate down` and then start the executable as usual._
